mountpoint-s3/src/fs.rs

```diff
@@ -604,15 +612,28 @@ where
         atime: Option<OffsetDateTime>,
         mtime: Option<OffsetDateTime>,
         _flags: Option<u32>,
+        size: Option<u64>,
     ) -> Result<Attr, Error> {
         tracing::info!(
-            "fs:setattr with ino {:?} flags {:?} atime {:?} mtime {:?}",
+            "fs:setattr with ino {:?} flags {:?} atime {:?} mtime {:?} size {:?}",
             ino,
             _flags,
             atime,
-            mtime
+            mtime,
+            size
         );
-        let lookup = self.superblock.setattr(&self.client, ino, atime, mtime).await?;
+        let lookup = match (self.superblock.setattr(&self.client, ino, atime, mtime).await, size) {
+            (Ok(lookup), _) => lookup,
+            (Err(e), Some(_)) => {
```

Comment:
I need to update it in comments. Actually, for the condition of overwrite, setattr is called with size having `Some` value, but for modifying metadata setattr is called with size `None`. So, I have used that to distinguish file overwrite message from modifying metadata in setattr.

Reply:
I think we need to crisp up the comment a bit.

What we're doing is ignoring all other arguments if size is set, to return a specific error that overwrite isn't supported. Maybe customer's application actually wants to set two things, one of which is size, and we'll just tell them to set allow overwrite to fix...

Maybe that's OK, but we need to consciously make that decision. (I need to think a bit more on whether this is the right thing to do, I don't have context on this decision.)

Reply:
Please also use `_size` so its clear to reader what we're ignoring without needing to look back up at the match.

Reply:
Here, I am trying to surface file overwrite error in the logs. At present, to overwrite a file, call `setatttr` is called first when `--allow-overwrite` flag is not used. So, `SetAttrNotPermittedOnRemoteInode` error is logged. 
In case of file overwrite, `setattr` is called size parameter having `Some` value. When we try to modify metadata we  have size as `None` in setattr. So, I was trying to use that to distinguish file overwrite error logging from general metadata modification on remote file error.
Note that I am only checking for size only  if we get the `SetAttrNotPermittedOnRemoteInode` error, so we do not falsely include other case of error.